sonar-plugin-server/src/main/java/jetbrains/buildserver/sonarplugin/sqrunner/ManageSQSActionController.java

```diff
@@ -70,14 +79,17 @@ private static String getAction(final @NotNull HttpServletRequest request) {
     public void process(final @NotNull HttpServletRequest request,
                         final @NotNull HttpServletResponse response,
                         final @Nullable Element ajaxResponse) {
-        if (ajaxResponse == null) {
+        final SProject project = getProject(request);
+        if (ajaxResponse == null || project == null) {
             return;
         }
 
-        final SProject project = getProject(request);
-        if (project == null) {
+        // Security test (user without management permission could access this controller)
+        if (!AuthUtil.hasGlobalPermission(securityContext.getAuthorityHolder(), Permission.CHANGE_SERVER_SETTINGS)){
```

Comment:
`CHANGE_SERVER_SETTINGS` usually means global server admin which has too many rights. As SonarQube Server is project-level configuration `EDIT_PROJECT` is much more suitable permission for this issue. `AuthUtil.hasPermissionToManageProject(securityContext.getAuthorityHolder(), project)` could be used.